summoner-cli/src/Summoner/Project.hs

```diff
@@ -197,16 +197,26 @@ generateProject isOffline projectName Config{..} = do
         stackMsg c = targetMessageWithText c "Stack" "used in this project"
 
 -- | Creates the directory and run GitHub commands.
-initializeProject :: Bool -> Settings -> IO ()
-initializeProject isOffline settings@Settings{..} = do
+initializeProject :: Settings -> IO ()
+initializeProject settings@Settings{..} = do
     createProjectDirectory settings
-    for_ (Map.toList settingsFiles) $ \(path, source) -> do
-        infoMessage $ "Creating extra file: " <> toText path
-        whenJustM (fetchSource isOffline source) (createFileWithParents path)
-
     when settingsGitHub $ doGithubCommands settings
     beautyPrint [bold, setColor Green] "\nJob's done\n"
 
+{- | This function fetches contents of extra file sources.
+-}
+fetchSources :: Bool -> Map FilePath Source -> IO [TreeFs]
+fetchSources isOffline = mapMaybeM sourceToTree . Map.toList
+  where
+    sourceToTree :: (FilePath, Source) -> IO (Maybe TreeFs)
+    sourceToTree (path, source) = do
+        infoMessage $ "Fetching content of the extra file: " <> toText path
+        fetchSource isOffline source >>= \case
+            Nothing -> do
+                errorMessage $ "Error fetching: " <> toText path
+                pure Nothing
```

Comment:
Is this correct, that in case of download (or just file getting) failure the process of the project generation will still continue without those files?

Reply:
@vrom911 I'm not sure, but implementing different logic is quite complicated. For example, if `isOffile` is `True` then all `Url` will fail. But maybe this is intentional and maybe I don't want things like `CONTRIBUTING.md` while developing locally :thinking:

Reply:
Yes, makes sense. By the way, where the mentioned case is handled?

Reply:
It's handled here:

https://github.com/kowainik/summoner/blob/38d1fceda39b8643c2860605a3eb175e387cd497/summoner-cli/src/Summoner/Source.hs#L65-L67

Reply:
Oh, right! Thank you :wine_glass: